codebuild/common-macos.sh

```diff
@@ -6,6 +6,25 @@ set -e
 set -x
 
 CMAKE_ARGS="$@"
+
+# ensure each required package is installed, if not, make a bottle for brew in ./packages
+# so it will be cached for future runs. If the cache is ever blown away, this will update
+# the packages as well
+# If the bottles are already in ./packages, then just install them
+function install_from_brew {
+    pushd ./packages
+    # usually the existing package is too old for one of the others, so uninstall
+    # and reinstall from the cache
+    brew uninstall --ignore-dependencies $1
+    if [ ! -e $1*bottle*.tar.gz ]; then
+        brew install --build-bottle $1
+        brew bottle --json $1
+        brew uninstall --ignore-dependencies $1
+    fi
+    brew install $1*bottle*tar.gz
```

Comment:
Should the `uninstall` be outside the if-block? Don't you want to uninstall/install regardless of whether the bottle exists in the cache?

Reply:
this is the original, and it builds fine so, I vote for letting it be.